vyper/semantics/analysis/module.py

```diff
@@ -150,15 +150,16 @@ def _compute_reachable_set(fn_t: ContractFunctionT, path: list[ContractFunctionT
     path = path or []
 
     path.append(fn_t)
-    root = path[0]
 
     for g in fn_t.called_functions:
         if g in fn_t.reachable_internal_functions:
             # already seen
             continue
 
-        if g == root:
-            message = " -> ".join([f.name for f in path])
+        if g in path:
+            cycle_start = path.index(g)
```

Comment:
i think printing out the full call trace will be easier for the user to debug

Reply:
addressed in: https://github.com/vyperlang/vyper/pull/4200/commits